Mathlib/Computability/PrimrecFilter.lean

```diff
@@ -0,0 +1,155 @@
+/-
+Copyright (c) 2025 David J. Webb. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: David J. Webb
+-/
+import Mathlib.Computability.Primrec
+
+/-!
+
+# Bounded quantifiers are primitive recursive
+This file contains lemmata for showing bounded existentially and universally quantified
+statements are primitive recursive, as well as more general filtering for arbitrary
+primcodable types.
+
+## Main results
+- Filtering for elements from a list that meet a primrec criterion is primrec
+- Checking whether a list has some element meeting a primrec criterion is primrec
+- Checking whether every element of a list meets a primrec criterion is primrec
+- Primitive recursive functions are closed under bounded existential and universal quantifiers
+
+## References
+- [R. I. Soare *Turing Computability - Theory and Applications*] [Soare2016]
+-/
+
+open List Primrec
+
+namespace Primrec
+
+variable {α} [Primcodable α] {f : α → Prop} [DecidablePred f]
+
+/-- Filtering a list for elements that satisfy a decidable predicate is primitive recursive. -/
+lemma filter (hf : PrimrecPred f) : Primrec fun L ↦ filter (f ·) L := by
```

Comment:
```suggestion
lemma listFilter (hf : PrimrecPred f) : Primrec fun L ↦ filter (f ·) L := by
```
seems more appropriate because of ambiguity with `Finset.filter`